app/main.py

```diff
@@ -1,2 +1,90 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.hash_table: list = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+
+        index = self.get_index_for_key(key)
+
+        if self.hash_table[index] is not None:
+            self.hash_table[index] = (key, hash(key), value)
```

Comment:
if the keys match you just have to update the value instead of overwriting everything